frontend/src/components/ColumnDataTable/index.vue

```diff
@@ -0,0 +1,249 @@
+    hasWorkspacePermissionV1(
+      "bb.permission.workspace.manage-sensitive-data",
+      currentUserV1.value.userRole
+    )
+  ) {
+    // True if the currentUser has workspace level sensitive data
+    // R+W privileges. AKA DBA or Workspace owner
+    return true;
+  }
+
+  // False otherwise
+  return false;
+});
+
+const columns = computed(() => {
+  const columns: (DataTableColumn<ColumnMetadata> & { hide?: boolean })[] = [
+    {
+      key: "name",
+      title: t("common.name"),
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return row.name;
+      },
+    },
+    {
+      key: "maskingLevel",
+      title: t("settings.sensitive-data.masking-level.self"),
+      hide: !showSensitiveColumn.value,
+      resizable: true,
+      width: 220,
+      render: (row) => {
+        return h(MaskingLevelCell, {
+          database: props.database,
+          schema: props.schema,
+          table: props.table,
+          column: row,
+          maskDataList: props.maskDataList,
+          readonly: !hasSensitiveDataPermission.value,
+        });
+      },
+    },
+    {
+      key: "semanticType",
+      title: t("settings.sensitive-data.semantic-types.self"),
+      hide: !showSensitiveColumn.value,
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return h(SemanticTypeCell, {
+          database: props.database,
+          schema: props.schema,
+          table: props.table,
+          column: row,
+          readonly: !hasSensitiveDataPermission.value,
+        });
+      },
+    },
+    {
+      key: "classification",
+      title: t("database.classification.self"),
+      hide: !showClassificationColumn.value,
+      width: 140,
+      render: (row) => {
+        return h(ClassificationLevelBadge, {
+          classification: row.classification,
+          classificationConfig: props.classificationConfig,
+        });
+      },
+    },
+    {
+      key: "type",
+      title: t("common.type"),
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return row.type;
+      },
+    },
+    {
+      key: "default",
+      title: t("common.default"),
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return getColumnDefaultValuePlaceholder(row);
+      },
+    },
+    {
+      key: "nullable",
+      title: t("database.nullable"),
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return row.nullable;
+      },
+    },
+    {
+      key: "characterSet",
+      title: t("db.character-set"),
+      hide: engine.value === Engine.POSTGRES,
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return row.characterSet;
+      },
+    },
+    {
+      key: "collation",
+      title: t("db.collation"),
+      hide: !showCollationColumn.value,
+      resizable: true,
+      width: 140,
+      render: (row) => {
+        return h(NEllipsis, null, { default: () => row.collation });
+      },
```

Comment:
May use `ellipsis: true` with `ellipsis-component: "ellipsis" or "performant-ellipsis"` for short

Reply:
Good catch!